macros/blueprint-proc-macro/src/sdk_main.rs

```diff
@@ -0,0 +1,103 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{Expr, ItemFn, Token};
+use syn::parse::{Parse, ParseStream};
+
+// This macro contains the following format:
+// #[sdk_main]
+// or
+// #[sdk_main(args...)]
+// all args are passed into the tokio::main attribute, with exception of certain keywords
+// all exception keywords are listed in mod kw below
+
+mod kw {
+    // Presented as #[sdk::main(env)]
+    syn::custom_keyword!(env);
+}
+
+// Add fields which are passed onto tokio, and those that are reserved for the sdk_main macro
+pub(crate) struct SdkMainArgs {
+    tokio_args: Option<Vec<Expr>>,
+    env: bool,
+}
+
+pub(crate) fn sdk_main_impl(args: &SdkMainArgs, input: &ItemFn) -> syn::Result<TokenStream> {
+    let tokio_args = if let Some(args) = &args.tokio_args {
+        quote! { ( #(#args),* ) }
+    } else {
+        quote! {}
+    };
+
+    let env_function_signature = if args.env {
+        quote! { env: gadget_sdk::config::GadgetConfiguration<gadget_sdk::parking_lot::RawRwLock> }
+    } else {
+        quote! {}
+    };
+
+    let env_passed_var = if args.env {
+        quote! { env }
+    } else {
+        quote! {}
+    };
+
+    // Next, we need to consider the input. It should be in the form "async fn main() { ... }"
+    // We must remove all the async fn main() and keep everything else
+    let input = input.block.clone();
+
+    let tokens = quote! {
+        use gadget_sdk::tokio;
+        #[tokio::main #tokio_args]
+        async fn main() {
+            gadget_sdk::logging::setup_log();
+            // Load the environment and create the gadget runner
+            let config: gadget_sdk::config::ContextConfig = gadget_sdk::structopt::StructOpt::from_args();
+            let env = gadget_sdk::config::load(config.clone()).expect("Failed to load environment");
```

Comment:
Why is there an optional `env` parameter to `sdk::main`? Is there a situation where you wouldn't want one? It gets loaded here anyway.

Being able to rename it would also be nice: `#[sdk::main(env = my_env)]`